### PR TITLE
Expose driving side per step

### PR DESCRIPTION
**Pull request by danpat**
# Issue

If you want to draw icons in a UI to represent a route, it can be helpful to know the driving side for a step so you can display the correct image.  

There are two use-cases this is aimed at:

  1. Drawing u-turn arrows.  We currently return `type=turn, maneuver=uturn`, and no other flag to indicate whether the uturn is ↶ or ↷.
  2. Drawing roundabout imagery.  In order to draw an image like the one describe [here](https://github.com/Project-OSRM/osrm-backend/issues/4178#issuecomment-310841761), you need to know whether to paint and count exits clockwise or counterclockwise.  You can currently make a *guess* from the intersection bearings, but it would be better to have an explicit flag.

This PR changes:
  1. Adds a new property to each `RouteStep` called `driving_side` with values of either `left` or `right`
  2. Bundles a GeoJSON file that roughly captures left-hand-side driving countries (I didn't do all the Carribean islands, and the land-border countries are pretty roughly drawn)

Open questions:
  1. Should the `driving_side` be a property of the `RouteStep` ?  A route may cross a border where the driving-side changes, and if there is no maneuver at the side-changing location, there would be no way to indicate it with the current scheme until the next maneuver is reached.  Maybe it's better to make it an `annotation`, or maybe a property of the `intersections` array?  Input appreciated.

This PR basically works - if the current design is acceptable, I'll add test cases and target this for OSRM 5.14.

The `driving_side.geojson` file could do with a bit of work - land borders are pretty rough, and I've missed a bunch of islands.  If anyone wants some straightforward polygon drawing task, this one is pretty straightforward :-)

/cc @oxidase 

## Tasklist
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [x] add regression / cucumber cases (see docs/testing.md)
 - [x] update API documentation
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

File: profiles/lib/way_handlers.lua [hunk 1]
```diff
@@ -563,7 +563,7 @@ function WayHandlers.driving_side(profile, way, result, data)
    elseif driving_side == 'right' then
       result.is_left_hand_driving = false
    else
-      result.is_left_hand_driving = profile.left_hand_driving
+      result.is_left_hand_driving = profile.properties.left_hand_driving
```
Comment (oxidase) [inline profiles/lib/way_handlers.lua]: Good catch! But  tests preconditions `profile.left_hand_driving = true` must be updated also

Comment (danpat) [reply profiles/lib/way_handlers.lua]: Done!

File: include/engine/guidance/assemble_steps.hpp [hunk 2]
```diff
@@ -204,22 +205,24 @@ inline std::vector<RouteStep> assembleSteps(const datafacade::BaseDataFacade &fa
...
+                      NO_ROTARY_NAME,
+                      duration / 10.,
+                      distance,
+                      weight / weight_multiplier,
+                      target_mode,
+                      maneuver,
+                      leg_geometry.FrontIndex(segment_index),
+                      leg_geometry.BackIndex(segment_index) + 1,
+                      {intersection},
+                      facade.IsLeftHandDriving(target_node_id)}); // TODO: this one is wrong
```
Comment (TheMarex) [inline include/engine/guidance/assemble_steps.hpp]: Why is it wrong?

File: package-lock.json [hunk 3]
```diff
@@ -1,8 +1,19 @@
 {
   "name": "osrm",
-  "version": "5.13.0-latest.1",
+  "version": "5.14.0-latest.1",
```
Comment (TheMarex) [inline package-lock.json]: We should not include the `package-lock.json` diff. Seems like this was accidentally committed?

File: include/engine/guidance/assemble_steps.hpp (hunk 2 above)
Comment (danpat) [reply include/engine/guidance/assemble_steps.hpp]: That comment is referring to itself (the comment is wrong) ;-)  Deleting.

**Pull request merged**